lib/vm-agent.js

```diff
@@ -9,450 +9,696 @@
- *    { uuid: '70ac24a6-962a-4711-92f6-6dc6a53ea59e',
- *       owner_uuid: '930896af-bf8c-48d4-885c-6573a94b1853',
- *       quota: 25,
- *       max_physical_memory: 128,
- *       zone_state: 'running',
- *       state: 'running',
- *       brand: 'joyent-minimal',
- *       cpu_cap: 100,
- *       last_modified: '2014-04-29T23:30:17.000Z' },
- *    '9181f298-a867-49c6-9f34-d57584e7047e':
- *     { uuid: '9181f298-a867-49c6-9f34-d57584e7047e',
- *       owner_uuid: '930896af-bf8c-48d4-885c-6573a94b1853',
- *       quota: 25,
- *       max_physical_memory: 1024,
- *       zone_state: 'running',
- *       state: 'running',
- *       brand: 'joyent-minimal',
- *       cpu_cap: 300,
- *       last_modified: '2014-04-29T23:36:33.000Z' } },
- *       ...
- *       ...
- *   }
- */
+    return (newVmobj);
+}
 
-VmAgent.prototype.updateSample = function (options, callback) {
+VmAgent.prototype.initialUpdate = function initialUpdate(callback) {
     var self = this;
-    var log = this.log;
-    var uuid = options.uuid;
-
-    var newSample;
-
-    function lookup(cb) {
-        // If we fail a lookup, newSample gets reset on every retry
-        var query;
-        var searchOpts = {};
-
-        newSample = {};
-
-        if (uuid) {
-            searchOpts.uuid = uuid;
-            query = 'uuid=' + uuid;
-        } else {
-            query = 'uuid=*';
-        }
+    var vms = {};
+
+    vasync.pipeline({arg: {}, funcs: [
+        function _getVmapiVms(stash, cb) {
+            var startLookup = (new Date()).getTime();
+
+            self.vmapiClient.getVms(self.server_uuid,
+                function _getVmsCb(err, vmobjs) {
+                    var doneLookup = (new Date()).getTime();
+                    var vmIdx;
+
+                    self.log.debug({
+                        action: 'VMAPI.lookup',
+                        elapsed: (doneLookup - startLookup),
+                        err: err,
+                        vmCount: (vmobjs ? vmobjs.length : 0)
+                    }, 'completed VMAPI.lookup() for VmAgent() init');
+
+                    if (err) {
+                        cb(err);
+                        return;
+                    }
+
+                    stash.vmapiVms = {
+                        compareVms: {},
+                        fullVms: {}
+                    };
+
+                    // vmobjs is an array of VM objects
+                    for (vmIdx = 0; vmIdx < vmobjs.length; vmIdx++) {
+                        stash.vmapiVms.fullVms[vmobjs[vmIdx].uuid]
+                            = vmobjs[vmIdx];
+                        stash.vmapiVms.compareVms[vmobjs[vmIdx].uuid]
+                            = makeComparable(vmobjs[vmIdx],
+                            self.comparisonFields, 'vmapi');
+                    }
+                    cb();
+                }
+            );
+        }, function _startWatcher(stash, cb) {
+            /*
+             * Just before we do the vmadm lookup we start the watcher which
```

Comment:
Nevermind, saw [the call to `self.watcher.stop` below](https://github.com/joyent/sdc-vm-agent/pull/5/files#diff-16538bdf89a8728b30d7f22d3878d453R631).